tree/ntuple/v7/src/RDaos.cxx

```diff
@@ -19,7 +19,24 @@
 #include <numeric>
 #include <stdexcept>
 
+namespace {
+struct RDaosRAII {
+   RDaosRAII() { daos_init(); }
+   ~RDaosRAII() { daos_fini(); }
+};
+static std::unique_ptr<RDaosRAII> _RDaosRAII;
+} // anonymous namespace
+
+
+////////////////////////////////////////////////////////////////////////////////
+
+
 ROOT::Experimental::Detail::RDaosPool::RDaosPool(std::string_view poolUuid, std::string_view serviceReplicas) {
+   {
+      if (!_RDaosRAII)
+         _RDaosRAII = std::make_unique<RDaosRAII>();
```

Comment:
Could this line be executed by several thread at the same time? (I.e. it currently is not thread safe).

Reply:
Fixed, thanks.  Changed this to a static local variable.  ISO C++ guarantees that these are _thread-safe_ initialized the first time the function is entered.